### PR TITLE
Rename modules

### PR DESCRIPTION
**Pull request by DemianTinkiel**
Refactoring module names for better clarity

* xacml-sdk-admin => xacml-sdk-pap-rest-impl
* xacml-sdk-api => xacml-sdk-pdp-api
* xacml-sdk-core => xacml-sdk-pdp-rest-impl
* Add README.md section with PAP api information

### CODE REVIEW

**Comment by DemianTinkiel** [comment]
Fixes #17 
Relates to #18

**Comment by DemianTinkiel** [comment]
@hargathor I was looking at existing examples and I see that they don't compile. Mostly out of referring to old `com.thalesgroup.authzforce.*` packages but also because the `PdpService` wont start. 


So I was thinking maybe we can have maven or the examples start a docker container with authzforce-ce-server directly? definitely required for the admin examples but would also benefit the other samples. What do you think?

**Comment by cdanger** [comment]
OK good idea to use docker to test the integration between the SDK and AuthzForce Server in general anyway. For this you can/should use the our docker image `authzforce/server:release-8.1.0` on Docker hub.

**Comment by DemianTinkiel** [comment]
@cdanger Done.  The library I'm using ([Test Containers](https://www.testcontainers.org))  supports being used as a JUnit rule. However I think the tests have very specific conditions, so instead I would suggest using [wiremock](http://wiremock.org/) that supports doing a MiM approach where you record the responses against the real thing and then returns canned responses (see tests on the `xacml-sdk-pap-rest-impl` for example).

Also take into account for automated tests in jenkins that downloading the docker image takes a while. Its fine for manual tests in your PC but clearing the workspace in between tests in jenkins would make it slow (unless you run a remote docker-machine Jenkins can contact)

File: authzforce-xacmlsdk-samples/src/main/java/org/ow2/authzforce/sdk/pap/SimplePolicyCRUD.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        domains = client.getDomains();
+        log.info("{} domains found: {}", domains.size(), domains.stream().map(Link::getHref).toArray());
+
+        DomainProperties domainProperties = client.getDomain(newDomainID);
+        log.info("Friendly domain name: {}", domainProperties.getExternalId());
+
+        log.error("(U)pdate");
+        client.addDomain("test domain", DOMAIN);
+
+        log.error("(D)elete... lets leave it till the end, if we delete the domain now we cant to policy CRUD on it");
```
Comment (hargathor) [inline authzforce-xacmlsdk-samples/src/main/java/org/ow2/authzforce/sdk/pap/SimplePolicyCRUD.java]: log.error("(D)elete... lets leave it till the end, if we delete the domain now we cant do policy CRUD on it");

Comment (hargathor) [reply authzforce-xacmlsdk-samples/src/main/java/org/ow2/authzforce/sdk/pap/SimplePolicyCRUD.java]: should be a typo

Comment (DemianTinkiel) [reply authzforce-xacmlsdk-samples/src/main/java/org/ow2/authzforce/sdk/pap/SimplePolicyCRUD.java]: Done

**Pull request merged**